ext/gd/gd.c

```diff
@@ -4282,12 +4282,28 @@ PHP_FUNCTION(imageresolution)
 	im = php_gd_libgdimageptr_from_zval_p(IM);
 
 	if (!res_x_is_null && !res_y_is_null) {
+		if (ZEND_SIZE_T_UINT_OVFL(res_x)) {
+			zend_argument_value_error(2, "must be lower than %u", UINT_MAX);
+			RETURN_THROWS();
+		}
+		if (ZEND_SIZE_T_UINT_OVFL(res_y)) {
+			zend_argument_value_error(3, "must be lower than %u", UINT_MAX);
+			RETURN_THROWS();
+		}
```

Comment:
Doesn't seem like we are checking that the value is positive? Which is a pre-existing issue.